util/perl/OpenSSL/Util.pm

```diff
@@ -126,6 +127,85 @@ sub quotify_l {
     } @_;
 }
 
+=over 4
+
+=item fixup_cmd_elements LIST
+
+Fixes up the command line elements given by LIST in a platform specific
+manner.
+
+The result of this function is a copy of LIST with strings where quotes and
+escapes have been injected as necessary depending on the content of each
+LIST string.
+
+This can also be used to put quotes around the executable of a command.
+I<This must never ever be done on VMS.>
+
+=back
+
+=cut
+
+sub fixup_cmd_elements {
+    # A formatter for the command arguments, defaulting to the Unix setup
+    my $arg_formatter =
+        sub { $_ = shift;
+              ($_ eq '' || /\s|[\{\}\\\$\[\]\*\?\|\&:;<>]/) ? "'$_'" : $_ };
+
+    if ( $^O eq "VMS") {        # VMS setup
+        $arg_formatter = sub {
+            $_ = shift;
+            if ($_ eq '' || /\s|["[:upper:]]/) {
+                s/"/""/g;
+                '"'.$_.'"';
+            } else {
+                $_;
+            }
+        };
+    } elsif ( $^O eq "MSWin32") { # MSWin setup
+        $arg_formatter = sub {
+            $_ = shift;
+            if ($_ eq '' || /\s|["\|\&\*\;<>]/) {
+                s/(["\\])/\\$1/g;
+                '"'.$_.'"';
+            } else {
+                $_;
+            }
+        };
+    }
+
+    return ( map { $arg_formatter->($_) } @_ );
+}
+
+=over 4
+
+=item fixup_cmd LIST
+
+This is a sibling of fixup_cmd_elements() that expects the LIST to be a
+complete command line.  It does the same thing as fixup_cmd_elements(),
+expect that it treats the first LIST element specially on VMS.
+
+=back
+
+=cut
+
+sub fixup_cmd {
+    return quotify_cmd_elements(@_) unless $^O eq 'VMS';
```

Comment:
fixup_cmd_elements - this is the test failure cause

Reply:
Fixed